lib/services/image.ts

```diff
@@ -0,0 +1,63 @@
+import { s3 } from './s3';
+import sharp from 'sharp';
+import { v4 } from 'uuid';
+
+const AWS_S3_BUCKET_NAME = process.env.AWS_BUCKET_NAME_BOOKBAZAR as string;
+const IMAGE_WIDTH = 300;
+const IMAGE_HEIGHT = 400;
+
+export async function uploadImage(image: string): Promise<string>
+{
+  const buffer = await resizeImage(image);
+  const key: string = v4()
+  const params = {
+    Body: buffer, 
+    Bucket: AWS_S3_BUCKET_NAME, 
+    Key: key, 
+    ServerSideEncryption: 'AES256', 
+    StorageClass: 'STANDARD_IA',
+    ContentEncoding: 'base64',
+    ContentType: 'image/png'
+  };
+
+  return new Promise(function(resolve, reject) {
+    s3.putObject(params, function(err, data){
+      if (err) {
+        reject(err);
+      }
+      else {
+        const link = `https://${AWS_S3_BUCKET_NAME}.s3.us-east-2.amazonaws.com/${key}`;
```

Comment:
Is there somewhere we can get this dynamically so that we don't have to hardcode this URL?

Reply:
I added the region as a variable - wasn't able to find a better way to get this link unfortunately

Reply:
Apparently you don't even need the region in the URL (e.g. https://book-bazar-images.s3.amazonaws.com/2127bde4-7c95-4d0b-9578-2b535addd630)